File: example-jvm/src/test/java/de/p7s1/qa/sevenfacette/utils/FileLoaderTest.java
package de.p7s1.qa.sevenfacette.utils;

import java.io.BufferedReader;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.stream.Collectors;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;


class FileLoaderTest {

  static final String RESOURCES_TEST_FOLDER = "testfiles/";

  @Test
  @Disabled
  void readRessourceFileAsInputStream() {
    InputStream inputStream = new FileLoader().loadFileFromResourceAsStream(RESOURCES_TEST_FOLDER, "test.txt");

    String result = new BufferedReader(new InputStreamReader(inputStream)).lines()
      .parallel().collect(Collectors.joining("\n"));

    assert(!result.isEmpty());
  }

  @Test
  @Disabled
  void readRessourceFileAsString() {
    String fileContent = new FileLoader().loadFileFromResourceAsString(RESOURCES_TEST_FOLDER, "test.txt");
    assert(!fileContent.isEmpty());
  }
}

File: core/src/jvmTest/kotlin/utils/FileLoaderTest.kt
package utils

import de.p7s1.qa.sevenfacette.utils.FileLoader
import org.junit.Test
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.stream.Collectors
import kotlin.test.assertFailsWith
import kotlin.test.assertNotNull

/**
 * Testingclass for [FileLoader].
 *
 * Testcases:
 * - load file
 * - file not empty
 *
 * @author Stella Bastug
 */
class FileLoaderTest {

    private val RESOURCES_TEST_FOLDER = "testfiles/"

    @Test
    fun fileStreamIsNotEmpty(){
        //arrange
        val loader = FileLoader()
        val inputStream = loader.loadFileFromResourceAsStream(RESOURCES_TEST_FOLDER, "test.txt")
        //act
        val result = BufferedReader(InputStreamReader(inputStream!!)).lines()
                .parallel().collect(Collectors.joining("\n"))
        //assert
        assert(result.isNotEmpty())
    }

    @Test
    fun readFileFromResourceAsStream(){
        //arrange
        val loader = FileLoader()
        //act
        val inputStream = loader.loadFileFromResourceAsStream(RESOURCES_TEST_FOLDER, "test.txt")
        //assert
        assertNotNull(inputStream)
    }

    @Test
    fun canNotReadResourceFileAsString() {
        //arrange
        val loader = FileLoader()
        // act and assert
        assertFailsWith<Exception> {
            loader.loadFileFromResourceAsString(RESOURCES_TEST_FOLDER, "tesst.txt")
        }
    }

    @Test
    fun fileStringIsNotEmpty() {
        //arrange
        val loader = FileLoader()
        //act
        val fileContent = loader.loadFileFromResourceAsString(RESOURCES_TEST_FOLDER, "test.txt")
        //assert
        assert(fileContent!!.isNotEmpty())
    }

}